signac_dashboard/views.py

```diff
@@ -62,41 +74,51 @@ def show_job(dashboard, jobid):
         g.jobs = dashboard._get_job_details([job])
         g.title = g.jobs[0]["title"]
         g.subtitle = g.jobs[0]["subtitle"]
+        session["context"] = "JobContext"
         return dashboard._render_job_view(default_view="grid")
 
 
-def get_file(dashboard, jobid, filename):
-    try:
-        job = dashboard.project.open_job(id=jobid)
-    except KeyError:
-        abort(404, "The job id requested could not be found.")
+def get_file(dashboard, filename, jobid=None):
+    if jobid is not None:
+        try:
+            job_or_project = dashboard.project.open_job(id=jobid)
+        except KeyError:
+            abort(404, "The job id requested could not be found.")
+    else:
+        job_or_project = dashboard.project
+    if job_or_project.isfile(filename):
+        directory = (
+            job_or_project.root_directory()
+            if jobid is None
+            else job_or_project.workspace()
+        )
+        mimetype = None
+        cache_timeout = 0
+        # Return logs as plaintext
+        textfile_regexes = ["job-.*\\.[oe][0-9]*", ".*\\.log", ".*\\.dat"]
+        for regex in textfile_regexes:
+            if re.match(regex, filename) is not None:
+                mimetype = "text/plain"
+        return send_from_directory(
+            directory,
+            filename,
+            mimetype=mimetype,
+            cache_timeout=cache_timeout,
+            conditional=True,
+        )
     else:
-        if job.isfile(filename):
-            mimetype = None
-            cache_timeout = 0
-            # Return logs as plaintext
-            textfile_regexes = ["job-.*\\.[oe][0-9]*", ".*\\.log", ".*\\.dat"]
-            for regex in textfile_regexes:
-                if re.match(regex, filename) is not None:
-                    mimetype = "text/plain"
-            return send_from_directory(
-                job.workspace(),
-                filename,
-                mimetype=mimetype,
-                cache_timeout=cache_timeout,
-                conditional=True,
-            )
-        else:
-            abort(404, "The file requested does not exist.")
+        abort(404, "The file requested does not exist.")
 
 
 def change_modules(dashboard):
     enabled_modules = set(session.get("enabled_modules", []))
+    context = session.get("context")
```

Comment:
Does this need a default value of `"JobContext"`? I assume it defaults to `None`.